src/cache.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::HashMap;
+use std::fmt;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::Read;
+use std::io::{self, Write};
+use std::sync::RwLock;
+
+pub struct SimpleStringCache {
+    pub cache: RwLock<HashMap<String, String>>,
+    pub file_path: String,
+}
+
+impl fmt::Debug for SimpleStringCache {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let cache = self.cache.read().unwrap();
+        f.debug_struct("SimpleCache")
+            .field("file_path", &self.file_path)
+            .field("cache", &*cache)
+            .finish()
+    }
+}
+
+impl SimpleStringCache {
+    pub fn new(file_path: String) -> io::Result<Self> {
+        let mut cache = HashMap::new();
+
+        Ok(SimpleStringCache {
+            cache: RwLock::new(cache),
+            file_path,
+        })
+    }
+
+    pub fn put(&self, key: String, value: String) -> io::Result<()> {
+        {
+            let mut cache = self.cache.write().unwrap();
+            cache.insert(key.clone(), value);
+        }
+        Ok(())
+    }
+
+    pub fn get(&self, key: &str) -> Option<String> {
+        let cache = self.cache.read().unwrap();
+        cache.get(key).cloned()
+    }
+
+    pub fn load_from_file(&self) -> io::Result<()> {
+        let mut file = File::open(&self.file_path)?;
```

Comment:
If the file doesn't exist, it will cause a panic at the caller side.

Reply:
Thank you!